### PR TITLE
#160 Paths endpoint

### LINKED ISSUE #160: Paths endpoint
ℹ️ This is part of the Epic #158 

## Context
Paths provide information about potential path payments. A path can be used to populate the necessary fields for a path payment operation.

https://developers.stellar.org/api/aggregations/paths/

## Objective
* To get access to all the aggregations exposed by Horizon.

## Checklist
- [x] List Strict Receive Payment Paths
- [x] List Strict Send Payment Paths

### PR DESCRIPTION
**Pull request by MrJuanGaviriaK**
Closes #160 

## Context
Paths provide information about potential path payments. A path can be used to populate the necessary fields for a path payment operation.

https://developers.stellar.org/api/aggregations/paths/

## Objective
* To get access to all the aggregations exposed by Horizon.

## Checklist
- [x] List Strict Receive Payment Paths
- [x] List Strict Send Payment Paths

### CODE REVIEW

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]
@MrJuanGaviriaK Thank you very much for contributing, you did an outstanding job 🚀. Some comments need to be addressed :)

**Comment by MrJuanGaviriaK** [comment]
@juanhurtado10 Done, thanks!!!

File: README.md [hunk 1]
```diff
@@ -663,6 +663,46 @@ Stellar.Horizon.FeeStats.retrieve()
...
+#### List strict receive payment paths
+- [Required] destination_asset_type. The type for the destination asset, **native**, **credit_alphanum4**, or **credit_alphanum12**.
+- [Required] destination_amount. The amount of the destination asset that should be received.
+- [Optional] source_account. The Stellar address of the sender.
+- [Optional] source_assets. A comma-separated list of assets available to the sender.
+- [Optional] destination_asset_issuer. The Stellar address of the issuer of the destination asset. Required if the **destination_asset_type** is not **native**
+- [Optional] destination_asset_code. The code for the destination asset. Required if the **destination_asset_type** is not **native**.
+
+```elixir
+Stellar.Horizon.PaymentPaths.list_receive_paths(destination_asset_type: "native", destination_amount: 5, source_account: "GBTKSXOTFMC5HR25SNL76MOVQW7GA3F6CQEY622ASLUV4VMLITI6TCOO")
```
Comment (juanhurtado10) [inline README.md:689]: @MrJuanGaviriaK please use atoms for the `destination_asset_type: :native`

File: lib/horizon/payment_paths.ex [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+                                 destination_asset_issuer: "GA654JC6QLA3ZH4O5V7X5NPM7KEWHKRG5GJA4PETK4SOFBUJLCCN74KQ"
+                                )
+    {:ok, %Paths{records: [%Path{}, ...]}}
+  """
+
+  @spec list_paths(args :: args()) :: response()
+  def list_paths(args \\ []) do
+    :get
+    |> Request.new(@endpoint)
+    |> Request.add_query(args, extra_params: allowed_query_options(:opt1))
```
Comment (juanhurtado10) [inline lib/horizon/payment_paths.ex:66]: @MrJuanGaviriaK it is not really clear what `opt1` means. Could you please use declarative names instead?

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]
@MrJuanGaviriaK wow amazing job done 🚀.

Two minor changes need to be applied before moving on.

File: lib/horizon/payment_paths.ex (hunk 2 above)
Comment (MrJuanGaviriaK) [reply lib/horizon/payment_paths.ex:66]: I'm going to fix it, thanks!!!

**Review by juanhurtado10 — APPROVED** [review]
@MrJuanGaviriaK There you go!! Thanks for your contribution!!! 🚀

**Pull request merged**